proposals/0025-dml-shader-correctness-test.md

```diff
@@ -0,0 +1,79 @@
+<!-- {% raw %} -->
+
+# DML Correctness Test
+
+* Proposal:
+  [0025-dml-shader-correctness-test](0025-dml-shader-correctness-test.md)
+* Author(s): [Sumit Agarwal](https://github.com/sumitsays)
+* Status: **Design In Progress**
+
+
+
+## Introduction
+
+Test suite to test the compute shader capability for the new hlsl compiler
+`Clang` is DirectML shaders. There will be 3 test stages:
+- Compilation test
+- Validation test
+- Correctness test   
+
+This document outlines the strategy to test the correctness of DirectML shaders
+against Clang. 
+
+## Background
+The relationship between DirectML shaders and DirectML API is *many-to-many*
+i.e. each DirectML shader is mapped to one or more DirectML public operator API
+and vice-versa. Aditionally, DirectML has TAEF based test to test each of their
+operator API via `DirectML.Test.OperatorTests.dll`.    
+Therefore we can leverage DirectML operator test infrastructure to test the
+correctness of DirectML shaders against Clang.
+
+### DirectML Shader Compilation
+DirectML compiles its shader offline and, concatenates and compresses all
+generated binary formatted DXIL (or DXBC) into a single compressed binary blob.
+The offline tool which performs all of these is called DirectMLShaderCompiler.
+It is a C++ based tool which internally uses DXC (or FXC) library to generate
+DXIL (or DXBC). Below flow diagram illustrates the working of the tool.
+ 
+![alt text](image.png)
+
+DirectML generates DXIL (or DXBC) for 3 platforms Windows, Linux and WSL using
+DirectMLShaderCompiler tool. During runtime DirectML picks the appropriate DXIL
+binary from the single blob and de-compress it.
+
+## Proposed solution
+Make DirectML use Clang and run the `DirectML.Test.OperatorTests.dll`. In order
+to make DirectML use new compiler we need to modify DMLShaderCompiler tool to
+avoid changing the basic building block of how DirectML consumes hlsl file. We
+will be following below strategy to modify the tool:
+- `ShaderParser` and `ShaderExpander` can be skipped because DirectML checks in
+  the expanded hlsl under the path `Product/Shaders/Generated`. We will be using
+  all the hlsl files present under this path. 
+- `ShaderCompiler` can also be skipped because we will be compiling all shaders
+  using Clang outside of this tool and will be placing the compiled DXIL into a
+  new path `Product/Shaders/Generated/Clang`.
+- Point DMLShaderCompiler to the new compiled DXIL path and perform rest of the
+  steps as it is.
+
+### Work Items
+- Introduce a new dummy platform `Clang` in DirectML which will use the DXIL
+  generated from Clang. This will give the flexibility to compare the result of
+  DXC and Clang from the same DirectML ADO branch.
+- Modify DMLShaderCompiler to work for the new platform `Clang` based on the
```

Comment:
Can you expand the details on what needs to be modified? I'm assuming modifications start in the ShaderCompressor but if thats the case lets explicitly state it.